ambari-common/src/main/python/resource_management/libraries/execution_command/execution_command.py

```diff
@@ -70,115 +75,241 @@ def get_value(self, query_string, default_value=None):
+    """
+    Retrieve java version from command.json, i.e "java_version": "8", note "8" will convert to 8
+    :return: an integer represents java version
+    """
+    from resource_management.libraries.functions.expect import expect_v2
+    return expect_v2("ambariLevelParams/java_version", int)
 
   def get_jce_name(self):
+    """
+    Retrieve jce name from command.json, i.e "jce_name": "jce_policy-8.zip"
+    :return: jce name string
+    """
     return self.__get_value("ambariLevelParams/jce_name")
 
   def get_db_driver_file_name(self):
+    """
+    Retrieve database driver file name, i.e "db_driver_filename": "mysql-connector-java.jar"
+    :return: DB driver name string
+    """
     return self.__get_value('ambariLevelParams/db_driver_filename')
 
   def get_db_name(self):
+    """
+    Retrieve database name, i.e Ambari server's db name is "db_name": "ambari"
+    :return: DB name
+    """
     return self.__get_value('ambariLevelParams/db_name')
 
   def get_oracle_jdbc_url(self):
+    """
+    Retrieve oracle database jdbc driver url,
+    i.e "oracle_jdbc_url": "http://c7302.ambari.apache.org:8080/resources//ojdbc6.jar"
+    :return: oracle jdbc url
+    """
     return self.__get_value('ambariLevelParams/oracle_jdbc_url')
 
   def get_mysql_jdbc_url(self):
+    """
+    Retrieve mysql database jdbc driver url,
+    i.e "mysql_jdbc_url": "http://c7302.ambari.apache.org:8080/resources//mysql-connector-java.jar"
+    :return: mysql jdbc url
+    """
     return self.__get_value('ambariLevelParams/mysql_jdbc_url')
 
   def get_agent_stack_retry_count(self):
-    return int(self.__get_value('ambariLevelParams/agent_stack_retry_count', 5))
+    """
+    Retrieve retry count for stack deployment on agent,
+    i.e "agent_stack_retry_count": "5"
+    :return: retry count for stack deployment on agent
+    """
+    from resource_management.libraries.functions.expect import expect_v2
+    return expect_v2('ambariLevelParams/agent_stack_retry_count', int, 5)
 
   def check_agent_stack_want_retry_on_unavailability(self):
+    """
+    Check if retry is needed when agent is not reachable
+    :return: True or False
+    """
     return self.__get_value('ambariLevelParams/agent_stack_retry_on_unavailability')
 
   def get_ambari_server_host(self):
+    """
+    Retrieve ambari server host url from command.json, i.e "ambari_server_host": "c7302.ambari.apache.org"
+    :return: ambari server url
+    """
     return self.__get_value("ambariLevelParams/ambari_server_host")
 
   def get_ambari_server_port(self):
+    """
+    Retrieve ambar server port number from command.json, i.e "ambari_server_port": "8080"
+    :return: amabari server port number
+    """
     return self.__get_value("ambariLevelParams/ambari_server_port")
 
   def is_ambari_server_use_ssl(self):
+    """
+    Check if ssl is needed to connect to ambari server
+    :return: True or False
+    """
     return self.__get_value("ambariLevelParams/ambari_server_use_ssl", False)
```

Comment:
Should we use expect_v2 here as well?

Reply:
at this time I only use expect_v2 if the old code uses expect(), but actually there is no standard to say where we should use expect() or now, the old code seems only to focus on expect(..., int)